src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -121,38 +133,51 @@ public PackageObjectFactory(String packageName, ClassLoader moduleClassLoader) {
      */
     @Override
     public Object createModule(String name) throws CheckstyleException {
-        Object instance = createObjectFromMap(name);
+        Object instance = null;
+        // if the name is a simple class name, try to find it in maps at first
+        if (!name.contains(".")) {
+            instance = createObjectFromMap(name, NAME_TO_FULL_MODULE_NAME);
+            if (instance == null) {
+                if (thirdPartyNameToFullModuleName == null) {
+                    thirdPartyNameToFullModuleName =
+                            generateThirdPartyNameToFullModuleName(moduleClassLoader);
```

Comment:
> it is already cached in field

My question is not about it being a cached field. I am specifically asking if we can have a public getter for `thirdPartyNameToFullModuleName` or the results of `generateThirdPartyNameToFullModuleName`.

Everything else except what i mentioned below is what I expected.